packages/cadl-python/src/emitter.ts

```diff
@@ -921,7 +921,11 @@ function emitStdScalar(scalar: Scalar & { name: IntrinsicScalarName }): Record<s
         case "plainDate":
             return { type: "date" };
         case "zonedDateTime":
-            return { type: "datetime", format: "date-time" };
+            if (location === "header") {
```

Comment:
left in the commenbt tp the isse but this should be handled with `getClientFormat` from typespec-client-generator-core.

Reply:
Since `getType` does not rely on `ModelProperty`, it is hard to add `getClientFormat` to it. I just added a new function to do the format changing.